docs/predictive-analytics/model-retraining/process.sql

```diff
@@ -0,0 +1,66 @@
+SET 'auto.offset.reset' = 'earliest';
+
+-- Create stream of predictions
+CREATE STREAM PREDICTED_WEIGHT(
+  "Fish_Id" VARCHAR KEY,
+  "Species" VARCHAR,
+  "Height" DOUBLE,
+  "Length" DOUBLE,
+  "Prediction" DOUBLE
+  )
+WITH(
+  KAFKA_TOPIC = 'kt.mdb.weight-prediction', 
+  VALUE_FORMAT = 'JSON',
+  PARTITIONS = 6
+);
+
+-- Create stream of actual weights
+CREATE STREAM ACTUAL_WEIGHT(
+  "Fish_Id" VARCHAR KEY,
+  "Species" VARCHAR,
+  "Weight" DOUBLE
+  )
+WITH(
+  KAFKA_TOPIC = 'kt.mdb.machine-weight', 
+  VALUE_FORMAT = 'JSON',
+  PARTITIONS = 6
+);
+
+-- Create stream joining predictions with actual weights
+CREATE STREAM DIFF_WEIGHT
+WITH(
+  KAFKA_TOPIC = 'diff_weight', 
+  VALUE_FORMAT = 'JSON'
+)
+AS SELECT
+   'Key' AS "Key",
```

Comment:
For my own education: what is `'Key'` ? Neither `prediction_weight` not `actual_weight` have a column called `'Key'` -- what does it refer to?

Reply:
From the blog post that this was taken from:  "Because we want to aggregate over all events and therefore need to provide a grouping column, we create an additional key field that always has the same value Key."

Reply:
Ah. -- Might need an explanation?

Sorry for not being able to read SQL... _hiding_

Reply:
I added a comment explaining the "Key" field.